### PR TITLE
Costmap_2d linting

### PR DESCRIPTION
**Pull request by DLu**
Done using the roslint command: 

`rosrun roslint cpplint --verbose=2 --filter=-runtime/references,-build/include_what_you_use,-readability/casting,-build/namespaces,-legal src/*cpp include/costmap_2d/*h plugins/*cpp`

This gets most of the lint reports. The main thing it doesn't like are the header directives that include the package name as well as the file name.

### CODE REVIEW

**Comment by mikeferguson** [comment]
I think it is actually the trailing underscore on the include guards that it does not like

File: costmap_2d/include/costmap_2d/costmap_2d_ros.h [hunk 1]
```diff
@@ -154,7 +154,7 @@ class Costmap2DROS
   /** @brief Returns the current padded footprint as a geometry_msgs::Polygon. */
   geometry_msgs::Polygon getRobotFootprintPolygon()
   {
-    return costmap_2d::toPolygon( padded_footprint_ );
+    return costmap_2d::toPolygon( padded_footprint_);
```
Comment (mikeferguson) [inline costmap_2d/include/costmap_2d/costmap_2d_ros.h]: the leading space on this doesn't trigger?

File: costmap_2d/include/costmap_2d/testing_helper.h [hunk 2]
```diff
@@ -9,19 +9,19 @@
...
+      costmap.setCost(j, i, map[index]);
     }
   }
 }
 
-char printableCost( unsigned char cost )
+char printableCost(unsigned char cost)
 {
-  switch( cost )
+  switch ( cost)
```
Comment (mikeferguson) [inline costmap_2d/include/costmap_2d/testing_helper.h]: leading space in front of cost?

File: costmap_2d/plugins/obstacle_layer.cpp [hunk 3]
```diff
@@ -182,7 +183,7 @@ void ObstacleLayer::onInitialize()
       boost::shared_ptr < message_filters::Subscriber<sensor_msgs::PointCloud>
           > sub(new message_filters::Subscriber<sensor_msgs::PointCloud>(g_nh, topic, 50));
 
-      if( inf_is_valid )
+      if ( inf_is_valid )
```
Comment (mikeferguson) [inline costmap_2d/plugins/obstacle_layer.cpp]: spaces around inf_is_valid

File: costmap_2d/plugins/obstacle_layer.cpp [hunk 4]
```diff
@@ -200,7 +201,7 @@ void ObstacleLayer::onInitialize()
       boost::shared_ptr < message_filters::Subscriber<sensor_msgs::PointCloud2>
           > sub(new message_filters::Subscriber<sensor_msgs::PointCloud2>(g_nh, topic, 50));
 
-      if( inf_is_valid )
+      if ( inf_is_valid )
```
Comment (mikeferguson) [inline costmap_2d/plugins/obstacle_layer.cpp]: spaces around inf_is_valid

File: costmap_2d/plugins/voxel_layer.cpp [hunk 5]
```diff
@@ -353,52 +353,52 @@ void VoxelLayer::raytraceFreespace(const Observation& clearing_observation, doub
...
-      //voxel_grid_.markVoxelLine(sensor_x, sensor_y, sensor_z, point_x, point_y, point_z);
+      // voxel_grid_.markVoxelLine(sensor_x, sensor_y, sensor_z, point_x, point_y, point_z);
       voxel_grid_.clearVoxelLineInMap(sensor_x, sensor_y, sensor_z, point_x, point_y, point_z, costmap_,
                                       unknown_threshold_, mark_threshold_, FREE_SPACE, NO_INFORMATION,
                                       cell_raytrace_range);
 
       updateRaytraceBounds(ox, oy, wpx, wpy, clearing_observation.raytrace_range_, min_x, min_y, max_x, max_y);
 
-      if( publish_clearing_points )
+      if ( publish_clearing_points )
```
Comment (mikeferguson) [inline costmap_2d/plugins/voxel_layer.cpp]: spaces around publish_clearing_points

File: costmap_2d/plugins/voxel_layer.cpp [hunk 6]
```diff
@@ -353,52 +353,52 @@ void VoxelLayer::raytraceFreespace(const Observation& clearing_observation, doub
...
         point.y = wpy;
         point.z = wpz;
-        clearing_endpoints_.points.push_back( point );
+        clearing_endpoints_.points.push_back(point);
       }
     }
   }
 
-  if( publish_clearing_points )
+  if ( publish_clearing_points )
```
Comment (mikeferguson) [inline costmap_2d/plugins/voxel_layer.cpp]: spaces around publish_clearing_points

File: costmap_2d/include/costmap_2d/costmap_2d_ros.h (hunk 1 above)
Comment (DLu) [inline costmap_2d/include/costmap_2d/costmap_2d_ros.h]: No, apparently not.

**Comment by DLu** [comment]
Trailing underscores are fine. 

### Valid
- COSTMAP_2D_H_
- COSTMAP_2D_H

### Invalid
- COSTMAP_COSTMAP_2D_H_
- COSTMAP_COSTMAP_2D_H
- COSTMAP_2D_COSTMAP_2D_H
- _COSTMAP_2D_H_
- COSTMAP_2D_H__
- COSTMAP_2D_
- COSTMAP_2D

**Comment by mikeferguson** [comment]
That seems like a (new) bug in roslint -- Might want to open a ticket about it

**Comment by DLu** [comment]
[Done.](https://github.com/ros/roslint/issues/30) Do we want to standardize the ifndef guards before merging?

**Comment by DLu** [comment]
Ok, ifndef guards standardized.

**Comment by DLu** [comment]
Closed in favor of #366.

**Pull request closed without merging**